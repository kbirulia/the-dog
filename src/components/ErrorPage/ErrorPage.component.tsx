import React, { FunctionComponent } from "react";
import "./ErrorPage.style.css";

export type ErrorPageProps = {
  title: string;
};

const ErrorPage: FunctionComponent<ErrorPageProps> = ({
  title,
  children,
}): JSX.Element => {
  return (
    <div className="error-page">
      <h3 className="main__title">{title}</h3>
      {children}
    </div>
  );
};

export default ErrorPage;
