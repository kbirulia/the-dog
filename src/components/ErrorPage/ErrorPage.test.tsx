import React from "react";
import renderer from "react-test-renderer";
import ErrorPage from "./ErrorPage.component";

jest.mock("../AllBreeds/AllBreeds.component");

describe("Error", () => {
  test("should render all breeds", () => {
    const tree = renderer.create(<ErrorPage title="Some title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
