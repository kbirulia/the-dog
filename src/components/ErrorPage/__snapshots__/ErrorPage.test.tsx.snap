// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render all breeds 1`] = `
<div
  className="error-page"
>
  <h3
    className="main__title"
  >
    Some title
  </h3>
</div>
`;
